### ISSUE TITLE
Warning while compiling dijkstra.c

### ISSUE THREAD

**Issue by singhpratyush** [iiitv/algos#273]
```c
dijkstra/dijkstra.c: In function ‘main’:
dijkstra/dijkstra.c:63:2: warning: passing argument 1 of ‘dijkstra’ from incompatible pointer type [enabled by default]
  printf("The minimum distance from %d to %d is %d\n", source, destination, dijkstra(graph, source, destination));
  ^
dijkstra/dijkstra.c:29:5: note: expected ‘const int (*)[8]’ but argument is of type ‘int (*)[8]’
 int dijkstra(const int graph[VERTICES][VERTICES], int souce, int destination) {
     ^
```

**Comment by Monal5031** [comment]
It compiles and executes successfully locally and even on ideone http://ideone.com/e.js/Z63EXj
![screenshot 13](https://user-images.githubusercontent.com/21009189/27088870-853f0054-5076-11e7-83e7-de111519a878.png)

**Comment by singhpratyush** [comment]
`gcc --version`?

**Comment by Monal5031** [comment]
I was just checking the version of `c` its `5.4.0`

**Comment by singhpratyush** [comment]
On Travis - 
```
gcc (Ubuntu 4.8.4-2ubuntu1~14.04.3) 4.8.4
Copyright (C) 2013 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

My local PC - 
```
gcc (Ubuntu 4.9.2-10ubuntu13) 4.9.2
Copyright (C) 2014 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by Monal5031** [comment]
Okay so that seems to be the problem then. So which version should we go with? `4.9.2` or above it?
Tried compiling on codechef ide got this error along with the warning you mentioned above.
```
prog.c:19:2: error: 'for' loop initial declarations are only allowed in C99 or C11 mode
  for (int i = 0; i < VERTICES; ++i) {
  ^
```
@singhpratyush Sir did you try compiling https://github.com/iiitv/algos/blob/master/rod_cutting_problem/rod_cutting.c it also should show the same warning.

**Comment by singhpratyush** [comment]
That requires `C11` mode (`-std=C11`).

**Comment by mohitkyadav** [comment]
@Monal5031 use C99 while compiling.

**Comment by Monal5031** [comment]
@mohitkyadav I know that, but as my local version of `C` is `5.4.0` I guess I don't need to do that.

**Comment by mohitkyadav** [comment]
No, I'm talking about the CodeChef ide.

**Comment by aviaryan** [comment]
Fixed in 40f0756
